src/Contract/Address.purs

```diff
@@ -121,37 +127,46 @@ import Ctl.Internal.Types.PubKeyHash
   ) as PubKeyHash
 import Ctl.Internal.Types.Scripts (StakeValidatorHash, ValidatorHash)
 import Ctl.Internal.Types.TypedValidator (TypedValidator)
-import Ctl.Internal.Types.UnbalancedTransaction
-  ( PaymentPubKey(PaymentPubKey)
-  ) as ExportUnbalancedTransaction
-import Data.Array (head)
+import Ctl.Internal.Types.UnbalancedTransaction (PaymentPubKey(PaymentPubKey)) as ExportUnbalancedTransaction
+import Data.Array (head, mapMaybe)
 import Data.Maybe (Maybe)
 import Data.Traversable (for, traverse)
 import Effect.Exception (error)
+import Prim.TypeError (class Warn, Text)
 
--- | Get the `Address` of the browser wallet.
--- TODO: change this to Maybe (Array Address)
--- https://github.com/Plutonomicon/cardano-transaction-lib/issues/1045
+-- | Get an `Address` of the browser wallet.
 getWalletAddress
-  :: forall (r :: Row Type). Contract r (Maybe Address)
-getWalletAddress = do
-  mbAddr <- wrapContract $ QueryM.getWalletAddresses <#> head
-  for mbAddr $
-    liftedM "getWalletAddress: failed to deserialize Address"
-      <<< pure
-      <<< toPlutusAddress
+  :: forall (r :: Row Type)
+   . Warn
+       ( Text
+           "This function returns only one `Adress` even in case multiple `Adresses` are available. Use getWalletAdresses instead"
+       )
+  => Contract r (Maybe Address)
+getWalletAddress = head <$> getWalletAddresses
+
+-- | Get all the `Addresses` of the browser wallet discarding errors.
+getWalletAddresses
+  :: forall (r :: Row Type). Contract r (Array Address)
+getWalletAddresses = do
+  addresses <- wrapContract $ QueryM.getWalletAddresses
+  pure $ mapMaybe toPlutusAddress addresses
```

Comment:
This is not the correct behavior. In case any of the addresses fails to convert, we must throw. Because we don't want the users to experience bugs that are impossible to debug.

Reply:
This applies to other functions as well.